### DISCUSSION TITLE
sea-orm-cli generate entity panic on table constraint parsing

### DISCUSSION THREAD

**Discussion by wbew** [SeaQL/sea-orm#743] [Q&A] [answered]
Hello! Opening this as a discussion instead of a bug report because I'm actually not sure if the issue is with a mistaken assumption in `SeaORM` or on my end in Postgres.

## Description

When I run `sea-orm-cli generate entity ...`, I encounter the following error during the `discover_constraints` phase.

```
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', /Users/will.wang/dev-projects/sea-schema/src/postgres/parser/table_constraints.rs:36:47
```

It seems to fail when a `Constraint::Check` type is found but the corresponding result does not have a `check_clause`. I was temporarily able to circumvent this issue by adding an `unwrap_or_default()`, but then I encountered the following error afterwards

```
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', /Users/will.wang/dev-projects/sea-schema/src/postgres/parser/table_constraints.rs:47:63
```

which seems to be the same as the one in #699. (Although the `sea-orm-cli` I have locally includes the bug fix for that issue).

I'm fairly new to all this database work, so excuse my naivety. Is this issue on SeaORM's side with improper `unwrap`/edge case handling or is it something inherent to our database? Unfortunately, I can't provide exact schema details. Thanks in advance!

(Note, I'm only interested in generating entities for 2 tables out of the hundreds in the database I'm working with. Thus, a couple `unwrap_or_default` hacked in locally seems to help this issue for now, but I'd love to understand a long term fix).

## Versions

sea-orm-cli: 0.8.1 (built locally on master at https://github.com/SeaQL/sea-schema/commit/00429f3bdbfe56af1afcee94b069a9619cea4263)
postgres: 9.6.22
os: MacOS Monterey 12.3.1

**Comment by billy1624** [comment, marked as answer]
Hey @001wwang, thanks for the report! I believe it has something to do with the version of PostgreSQL, we have tested SeaSchema on PostgreSQL vision 10+ but not version 9. Could you please provide a minimum reproducible schema?

**Reply by wbew** [reply]
I see. Sure thing, I will try to construct a reproducible example. It may be a while before I can dedicate time to it, so please bear with me.
